File: 657.robot-return-to-origin.go
package main

/*
 * @lc app=leetcode id=657 lang=golang
 *
 * [657] Robot Return to Origin
 */

// @lc code=start
func judgeCircle(moves string) bool {
	x, y := 0, 0

	for _, r := range moves {
		if 'U' == r {
			y += 1
		} else if 'D' == r {
			y -= 1
		} else if 'R' == r {
			x += 1
		} else {
			x -= 1
		}
	}
	return x == 0 && y == 0
}

// @lc code=end

File: 657.robot-return-to-origin.rb
#
# @lc app=leetcode id=657 lang=ruby
#
# [657] Robot Return to Origin
#

# @lc code=start
# @param {String} moves
# @return {Boolean}
def judge_circle(moves)
  x, y = 0, 0
  moves.each_char do |move|
    case move
    when 'U'
      y+=1
    when 'D'
      y-=1
    when 'R'
      x+=1
    when 'L'
      x-=1
    else
      raise NotInprementError
    end
  end

  return x == 0 && y == 0
end
# @lc code=end

